### DISCUSSION TITLE
Arch linux GLIBC not found but i have it installed?

### DISCUSSION THREAD

**Discussion by zephiowl** [cross-rs/cross#1025] [Q&A] [answered]
Hey so i'm getting this error whenever i try to CROSS compile my app for windows (it builds for linux fine?)

```
/target/release/build/futures-util-6548a22afc3d2d3e/build-script-build: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /target/release/build/futures-util-6548a22afc3d2d3e/build-script-build)
```
But whenever i yay -Qs it says i have glibc?
```
[xanthus@xanthuslinuxdesktop webhook_sender]$ yay -Qs glibc
local/glibc 2.36-4
    GNU C Library
local/lib32-glibc 2.36-4
    GNU C Library (32-bit)
```

I was wondering what may be causing this issue; as i wasn't having it a few days ago? And how i could go about fixing it?

**Comment by Alexhuszagh** [comment, marked as answer]
Have you tried running `cargo clean` and then rebuilding? It seems like this is unfortunately another casualty of #724. If that solves the issue, I can provide the explanation in depth.

**Reply by zephiowl** [reply]
Hey thanks i managed to fix the issue by moving the target dir; sadly cargo clean didn't seem to work but its not to big of an issue for me. Thanks a ton for helping me out.

**Reply by Emilgardis** [reply]
interesting, `cargo clean` with no arguments is just a straight `rm -rf target` basically


```
With no options, cargo clean will delete the entire target directory.
```

https://doc.rust-lang.org/cargo/commands/cargo-clean.html#description

great that you got it working now though!

**Reply by Alexhuszagh** [reply]
I'm very glad you got it working. We've mentioned a few options because we can't directly fix this issue, but the best solution is to export `CARGO_TARGET_DIR` as a path specific to the target. It's unfortunately, but setting some sort of build script being `./cross.sh build aarch64-unknown-linux-gnu` to be:

```bash
#!/usr/bin/env bash
cmd="${1}"
shift
target="${1}"
shift
cross "${cmd}" --target-dir "${target}/target" --target "${target}" "${@}"
```

Is probably the best solution. I wish there was more we could do. This way, every build script runs independently for each target, avoiding any bugs due to a toolchain changes not rerunning build scripts.
